src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/PdfGeneration/HealthCertificatePDFVersionViewController.swift

```diff
@@ -0,0 +1,124 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import UIKit
+import PDFKit
+import LinkPresentation
+
+class HealthCertificatePDFVersionViewController: DynamicTableViewController, DismissHandling, UIActivityItemSource {
+
+	// MARK: - Init
+
+	init(
+		viewModel: HealthCertificatePDFVersionViewModel,
+		onTapPrintPdf: @escaping (Data) -> Void,
+		onTapExportPdf: @escaping (PDFExportItem) -> Void,
+		onDismiss: @escaping () -> Void
+	) {
+		self.viewModel = viewModel
+		self.onTapPrintPdf = onTapPrintPdf
+		self.onTapExportPdf = onTapExportPdf
+		self.onDismiss = onDismiss
+
+		super.init(nibName: nil, bundle: nil)
+	}
+
+	@available(*, unavailable)
+	required init?(coder: NSCoder) {
+		fatalError("init(coder:) has not been implemented")
+	}
+
+	// MARK: - Overrides
+
+	override func viewDidLoad() {
+		super.viewDidLoad()
+
+		view = viewModel.pdfView
+		view.backgroundColor = .enaColor(for: .background)
+		
+
+		let printButton = UIBarButtonItem(image: UIImage(named: "Icons_Printer"), style: .plain, target: self, action: #selector(didTapPrintButton))
+		let shareButton = UIBarButtonItem(image: UIImage(named: "Icons_Share"), style: .plain, target: self, action: #selector(didTapShareButton))
+		
+		if UIPrintInteractionController.isPrintingAvailable {
+			navigationItem.rightBarButtonItems = [shareButton, printButton]
+		} else {
+			navigationItem.rightBarButtonItem = shareButton
+		}
+	}
+	
+	override func viewWillAppear(_ animated: Bool) {
+		super.viewWillAppear(animated)
+		navigationController?.navigationBar.prefersLargeTitles = false
+		// Must be set here, otherwise navBar will be translucent.
+		navigationController?.navigationBar.isTranslucent = false
+	}
+	
+	// MARK: - DismissHandling
+	
+	func wasAttemptedToBeDismissed() {
+		onDismiss()
+	}
```

Comment:
I wonder why we use `DismissHandling` here if there is no close button? Is it what UX wanted?

Reply:
Yeah, right, there should be no dismiss to match our pattern in the app. I will remove it.